PyMca5/PyMcaCore/TiledDataSource.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import numpy as np
+import os
+import time
+
+from PyMca5.PyMcaIO import TiledFile
+
+SOURCE_TYPE = 'Tiled'
+
+_logger = logging.getLogger(__name__)
+
+class TiledDataSource(object):
+    """
+    Creates instance of a Tiled Data Source class. This is neccesary
+    to create a Tiled Tab in the QDispatcher, which houses the Tiled
+    Browser.
+
+    This is largely based on the NexusDataSource class, but all Data 
+    Source tabs (Spec, EDF, SPS) have an analogous class.
+
+    See QDataSource.py
+    """
+
+    def __init__(self, nameInput):
+        self.nameList = nameInput
+        self.source_type = SOURCE_TYPE
+        self.__sourceNameList = self.__sourceNameList
+        self._sourceObjectList = []
+        self.refresh()
```

Comment:
I think I see a few problems here:
1. Need to verify that `nameInput` is a `list`.
2. self.__sourceNameList is self-defined (cyclic assignment)
3. self.NameList should be defined

Here is a suggestion, based on [`NexusDataSource`](https://github.com/NSLS2/pymca/blob/4711c8e762cb066b0ed729642a8d3899e80bd190/PyMca5/PyMcaCore/NexusDataSource.py#L211-L230).

```suggestion
    def __init__(self, nameInput):
        if isinstance(nameInput, list):
            nameList = nameInput
        else:
            nameList = [nameInput]
        self.sourceName = []
        self.source_type = SOURCE_TYPE
        self.__sourceNameList = self. self.sourceName
        self._sourceObjectList = []
        self.refresh()
```